src/core/server/ui_settings/settings/theme.ts

```diff
@@ -6,17 +6,45 @@
  * Side Public License, v 1.
  */
 
-import { schema } from '@kbn/config-schema';
+import { schema, Type } from '@kbn/config-schema';
 import { i18n } from '@kbn/i18n';
 import { UiSettingsParams } from '../../../types';
 
-export const getThemeSettings = (): Record<string, UiSettingsParams> => {
+function parseThemeTags() {
+  if (!process.env.KBN_OPTIMIZER_THEMES) {
+    return ['v8light', 'v8dark'];
+  }
+
+  if (process.env.KBN_OPTIMIZER_THEMES === '*') {
+    return ['v8light', 'v8dark', 'v7light', 'v7dark'];
+  }
+
+  return process.env.KBN_OPTIMIZER_THEMES.split(',').map((t) => t.trim());
+}
+
+interface GetThemeSettingsOptions {
+  isDist?: boolean;
+}
+
+export const getThemeSettings = (
+  options: GetThemeSettingsOptions = {}
+): Record<string, UiSettingsParams> => {
+  const isDist = options?.isDist ?? true;
+
+  const themeTags = parseThemeTags();
+  const availableVersions = ['v7', 'v8'].filter(
+    (v) => isDist || themeTags.some((t) => t.startsWith(v))
+  );
+
+  const defaultDarkMode = isDist ? false : themeTags[0].endsWith('dark');
+  const defaultVersion = isDist ? 'v8' : themeTags[0].slice(0, 2);
```

Comment:
It's more verbose, but I would just avoid calling `parseThemeTags` when `dist` is `true`

```ts
const getThemeInfos = (dist: boolean) => {
  if (dist) {
    return {
      defaultDarkMode: false,
      defaultVersion: 'v8',
      availableVersions: ['v7', 'v8'],
    };
  } else {
    const themeTags = parseThemeTags();
    return {
      availableVersions: ['v7', 'v8'].filter((v) => themeTags.some((t) => t.startsWith(v))),
      defaultDarkMode: themeTags[0].endsWith('dark'),
      defaultVersion: themeTags[0].slice(0, 2),
    };
  }
};

// in `getThemeSettings`

const { availableVersions, defaultDarkMode, defaultVersion } = getThemeInfos(options?.isDist ?? true)

```